gibbon-compiler/src/Gibbon/Passes/Codegen.hs

```diff
@@ -202,7 +202,8 @@ codegenProg cfg prg@(Prog info_tbl sym_tbl funs mtal) =
         let struct_tys = uniqueDicts $ S.toList $ harvestStructTys prg
         return ((L.nub $ makeStructs struct_tys) ++ prots ++
                 [gibTypesEnum, initInfoTable info_tbl, initSymTable sym_tbl] ++
-                funs' ++ [main_expr'])
+                funs' -- ++ [main_expr']
```

Comment:
This is one of two showstoppers in this patch afaics. Need to find a way to selectively block the generation of `main` if we run in the `Library` mode.

Reply:
I was working on a PR for this myself, but couldn't complete everything I wanted to.

There's a "Library" mode of compilation (passed in as a flag) that we can use here.

Reply:
Oh, no worries at all, there's now way you'd know that I was also doing this :D I'll push my commits to this branch when I get to my PC in the evening.

The two remaining issues I had were (1) calling `gib_init` somehow, and (2) the generated executable somehow couldn't find the `libgibbon_rts_ng.so` file. I don't recall if (2) was a problem from before or if it's a new thing.

Reply:
All right, this tiny concern about the generation of `main` should be addressed now. The other concerns I restated in the OP. So I "resolve" the thread...